### DISCUSSION TITLE
Wrong timezone

### DISCUSSION THREAD

**Discussion by bottiger** [dani-garcia/vaultwarden#3775] [Q&A]
For whatever reason Vaultwarden have started running with a wrong timezone. I live in UTC+1 and vaultwarden seems to think it's two hours behind.

This log entry shows the problem

`[2023-08-14 07:55:10.449][vaultwarden::api::core::two_factor::authenticator][ERROR] Invalid TOTP code! Server time: 2023-08-14 05:55:10 UTC IP: 77.233.250.50`

I have not done anything special to set the timezone of Vaultwarden specifically, but it worked previously and the server time is correct.

**Comment by BlackDex** [comment]
Please provide a Support String which you can generate by going to `/admin/diagnostics`

Also, Vaultwarden uses the timezone configured on either the host or by the environment variable `TZ`.
Sometimes people also pass `/etc/timezone` to the container, if you did that and it is incorrect then that can cause issues too.

**Comment by bottiger** [comment]
I'm sorry. In the meantime I did a reboot of the entire system and the problem fixed itself. Now the time  zones are correct, and I assume I'm unable to provide any meaningful debug info.

If you are still interested please let me know

**Comment by VarChar42** [comment]
I had the same problem after upgrading to 1.29.1.

Removing the "TZ" environment variable, so the container runs on UTC time, fixed the problem.

**Comment by CirKanus** [comment]
Imposible to activate 2FA because vaulltwarden ignore TZ variable.

https://github.com/truenas/charts/pull/1516#issuecomment-1710298107

Any help please.

**Reply by BlackDex** [reply]
Check `/admin/diagnostics`, and see if the date and time are correct there.

**Reply by CirKanus** [reply]
![Screenshot 2023-09-08 at 08-31-02 Vaultwarden Admin Panel](https://github.com/dani-garcia/vaultwarden/assets/113084123/5583e0f7-de9f-42fb-b32c-ebe8aabd8d40)

**Reply by CirKanus** [reply]
The server time is correct, no other else.

**Reply by BlackDex** [reply]
The server time is wrong. It isn't 08:xx in UTC right now, so you are doing something strange.
The NTP Time is the correct time, same as your browser.

It actually looks like your server/host is 2 hours upfront, or you are using the wrong TZ there.
This, because +02:00 is also not 10h right now.

This is mine, and my TZ is `Europe/Amsterdam`
![image](https://github.com/dani-garcia/vaultwarden/assets/610450/c3305d0e-7e4b-4a1b-b3db-c8ba251021c9)

**Reply by CirKanus** [reply]
In me case the problem where in the hardware clock, thanks for the help.
